server/server/Controllers/customerCrud.js

```diff
@@ -0,0 +1,64 @@
+const mongoose = require("mongoose");
+const User = require("../Models/user");
+
+const getCustomerById = async (req, res) => {
+  const { id } = req.params;
+
+  try {
+    const customer = await User.findById(id).populate("cart");
+
+    if (!customer) {
+      return res.status(404).json({ message: "Customer not found" });
+    }
+
+    res.json(customer);
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ message: error.message });
+  }
+};
+
+
+const updateCustomer = async (req, res) => {
+  const { id } = req.params;
+
+  try {
+    const updatedCustomer = await User.findByIdAndUpdate(
+      id,
+      req.body,
+      { new: true }
+    );
```

Comment:
we need to extract information from body, we cant trust what could be sent other than our fields.
also lets run validation rules before updating
```suggestion
    const updatedCustomer = await User.findByIdAndUpdate(
      id,
      <here we serialize corresponding fields from req.body>,
      { new: true, runValidators: true }
    );
```